meilisearch/index.py

```diff
@@ -153,3 +143,650 @@ def get_index(config, uid):
         if uid is not None:
             return Index(config, uid=uid)
         raise Exception('Uid is needed to find index')
+
+    def get_all_update_status(self):
+        """Get all update status from MeiliSearch
+
+        Returns
+        ----------
+        update: `list`
+            List of all enqueued and processed actions of the index.
+        """
+        return self.http.get(
+            '{}/{}/{}'.format(
+                self.config.paths.index,
+                self.uid,
+                self.config.paths.update
+            )
+        )
+
+    def get_update_status(self, update_id):
+        """Get one update from MeiliSearch
+
+        Parameters
+        ----------
+        update_id: int
+            identifier of the update to retrieve
+        Returns
+        ----------
+        update: `list`
+            List of all enqueued and processed actions of the index.
+        """
+        return self.http.get(
+            '{}/{}/{}/{}'.format(
+                self.config.paths.index,
+                self.uid,
+                self.config.paths.update,
+                update_id
+            )
+        )
+
+    def get_stats(self):
+        """Get stats of an index
+
+        Get information about the number of documents, field frequencies, ...
+        https://docs.meilisearch.com/references/stats.html
+        Returns
+        ----------
+        stats: `dict`
+            Dictionnary containing stats about the given index.
+        """
+        return self.http.get(
+            '{}/{}/{}'.format(
+                self.config.paths.index,
+                self.uid,
+                self.config.paths.stat,
+            )
+        )
+
+    # pylint: disable=dangerous-default-value
+    # Not dangerous because opt_params is not modified in the method
+    # See: https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python
+    def search(self, query, opt_params={}):
```

Comment:
In the case none is not a dangerous default value, could we not change this {} to a none like the other methods in this class and add the following check?

```
if opt_params is None:
            parameters = {}
```

Reply:
Yes! Done! :)